### PR TITLE
Feature/#1852 grand rework dashboard page store

### PR DESCRIPTION
**Pull request by aminbksi**

### CODE REVIEW

File: src/core/models/Content/Content.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {Instance, types} from 'mobx-state-tree';
+
+const Content = types
+  .model('Content', {
+    text: types.maybe(types.string),
+    title: types.maybe(types.string),
+    type: types.maybe(types.string),
+    url: types.maybe(types.string)
+  })
+  .views((self) => ({}));
```
Comment (PAKovalev) [inline src/core/models/Content/Content.ts:10]: empty

File: src/core/models/Tile/Tile.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    type: types.enumeration(Object.values(TileType)),
+    content: types.maybe(types.frozen<ContentInterface>()),
+    edited: types.maybe(types.number),
+    internal: types.maybe(types.boolean),
+    render: types.maybe(types.number),
+    updatedAt: types.maybeNull(types.string),
+    uiTypeId: UUIDModel,
+    spaceId: UUIDModel
+  })
+  .views((self) => ({}));
```
Comment (PAKovalev) [inline src/core/models/Tile/Tile.ts:22]: empty

File: src/core/models/TileList/TileList.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {cast, Instance, types} from 'mobx-state-tree';
+import {cloneDeep} from 'lodash-es';
+
+import {COLUMNS} from 'core/constants';
+
+import {Tile, TileInterface} from '../Tile';
```
Comment (PAKovalev) [inline src/core/models/TileList/TileList.ts:6]: from 'core/models'

File: src/core/models/TileList/TileList.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    status: types.maybe(types.number),
+    message: types.maybe(types.string)
+  })
+  .actions((self) => ({
+    updateTiles(tiles: TileInterface[]) {
+      self.tiles = cast(cloneDeep(tiles));
+    }
+  }))
+  .views((self) => ({
+    get tileMatrix() {
```
Comment (PAKovalev) [inline src/core/models/TileList/TileList.ts:21]: `get tileMatrix(): TileInterface[][] {`. I mean `return type`.

File: src/scenes/collaboration/pages/DashboardPage/DashboardPage.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  }, []);
+
+  const leaveCollaborationSpaceCall = useLeaveCollaborationSpace();
+  const {collaborationState, collaborationDispatch} = useCollaboration();
+  const stageModeLeave = useStageModeLeave(collaborationState.collaborationSpace?.id);
+
+  // TODO: make as action in store
+  const leaveCollaborationSpace = () => {
+    if (collaborationState.collaborationSpace) {
+      UnityService.triggerInteractionMsg?.(
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/DashboardPage/DashboardPage.tsx:41]: Make action under `UnityService`

File: src/scenes/collaboration/pages/DashboardPage/components/templates/Dashboard/components/TileItem/components/TileDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  videoUrl?: string;
+  imageUrl?: string;
+  provided?: DraggableProvided;
+}
+
+const TileDetail: FC<PropsInterface> = ({tile, videoUrl, imageUrl, provided}) => {
+  const theme = useTheme();
+  const {
+    collaborationStore: {spaceStore}
+  } = useStore();
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/Dashboard/components/TileItem/components/TileDetail.tsx:23]: Don't use store in this component. Make props.

File: src/scenes/collaboration/stores/DashboardStore/DashboardStore.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import {ResetModel, TileList} from 'core/models';
+
+import {Dashboard} from './models/Dashboard';
+
+const DashboardStore = types.compose(
+  ResetModel,
+  types
+    .model('DashboardStore', {
+      tileList: types.optional(TileList, {}),
+      dashboardStore: types.optional(Dashboard, {})
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/DashboardStore/DashboardStore.ts:12]: `dashboard: types.optional(Dashboard, {})`

File: src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+          {
+            spaceId
+          }
+        );
+        if (response) {
+          self.tileList = cast(response);
+          response.tiles.map((tile) => {
+            if (
+              !tile.edited &&
+              (tile.permanentType === PermanentType.POSTER ||
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts:34]: `[PermanentType.POSTER, PermanentType.MEME, PermanentType.DESCRIPTION, PermanentType.DESCRIPTION].includes(tile.permanentType) {}` will be smaller.

File: src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+          data: {...self.tileList, tiles: tiles}
+        });
+      }),
+      setImageUrl(tile: TileInterface) {
+        self.imageUrl = `${appVariables.RENDER_SERVICE_URL}/get/${
+          tile.permanentType === PermanentType.VIDEO ? tile.content?.url : tile.hash
+        }`;
+      },
+      setVideoUrl(tile: TileInterface) {
+        self.videoUrl = `https://www.youtube.com/embed/${youtubeVideoPath(
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts:55]: Make `youtubeVideoFullPath`

File: src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+          tile.content?.url ?? '',
+          tile.id
+        )}`;
+      },
+      setTilesWithPositions(tiles: TileInterface[][]) {
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts:60]: Return type

File: src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+              row: rowI,
+              id: t.internal ? undefined : t.id
+            }))
+          )
+          .flat();
+
+        self.tileList.updateTiles(tilesWithPositions);
+        this.updatePositions(tilesWithPositions);
+      },
+      onDragEnd({source, destination}: DropResult) {
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/DashboardStore/models/Dashboard/Dashboard.ts:75]: return type

File: src/shared/services/i18n/locales/en-gb.ts [hunk 12]
```diff
@@ -259,7 +261,8 @@ export const enGb = {
       create: 'create',
       stage: 'Stage mode',
       kickedFromMeeting: 'You have been kicked from a space',
-      joinGathering: 'Gathering Event is starting now'
+      joinGathering: 'Gathering Event is starting now',
+      updateSpace: 'UPDATE YOUR SPACE WITH SOME ENTICING CONTENT'
```
Comment (PAKovalev) [inline src/shared/services/i18n/locales/en-gb.ts:265]: Why do you uppercase? Do it using css

File: src/ui-kit/molecules/TileMenu/TileMenu.styled.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  border-radius: 20px;
+
+  transition-property: opacity;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-duration: 150ms;
+
+  mixblendmode: difference;
+`;
+
+export const SvgItem = styled(SvgButton)``;
```
Comment (PAKovalev) [inline src/ui-kit/molecules/TileMenu/TileMenu.styled.ts:25]: Don't mix styles.

**Review by PAKovalev — APPROVED** [review]

File: src/core/utils/image.utils.ts [hunk 14]
```diff
@@ -0,0 +1,37 @@
+import {useEffect, useState} from 'react';
+
+// @ts-ignore
```
Comment (Danchev1) [inline src/core/utils/image.utils.ts:3]: can we remove ts-ignore the method looks generic, what is the problem?

Comment (Danchev1) [inline src/core/utils/image.utils.ts:3]: can we remove ts-ignore the method looks generic, what is the problem?

Comment (Danchev1) [inline src/core/utils/image.utils.ts:3]: can we remove ts-ignore the method looks generic, what is the problem?

**Review by Danchev1 — CHANGES_REQUESTED** [review]

File: src/scenes/collaboration/pages/DashboardPage/components/templates/Dashboard/components/TileItem/components/TileDetail.tsx [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+  tile,
+  videoUrl,
+  imageUrl,
+  provided,
+  isAdmin,
+  isMember
+}) => {
+  const theme = useTheme();
+
+  const handleEdit = () => {
```
Comment (Danchev1) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/Dashboard/components/TileItem/components/TileDetail.tsx:30]: maybe you forgot them?

**Review by Danchev1 — CHANGES_REQUESTED** [review]

**Pull request merged**